library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
@leodemoura Can't we make the error message much better by checking the type of the returned pre-expression?  That is, when we evaluate the parser, we already know the type α.  We can then elaborate the returned pre-expression, and check whether it has indeed type α.